base/server/src/com/netscape/cms/profile/def/ServerKeygenUserKeyDefault.java

```diff
@@ -0,0 +1,486 @@
+            request.setExtData("isServerSideKeygen", "true");
+            CryptoToken token = cm.getInternalKeyStorageToken();
+
+            String keyTypeStr = request.getExtDataInString("keyType");
+            String keyType = "RSA";
+            int keySize = 2048;
+            String curveName = "nistp256";
+
+            // Populate the keyType and keySize/keyCurve
+
+            if (keyTypeStr != null && !keyTypeStr.isEmpty()) {
+                logger.debug("ServerKeygenUserKeyDefault: populate: keyType in request: " + keyTypeStr);
+                keyType = keyTypeStr;
+            } else {
+                logger.debug("ServerKeygenUserKeyDefault: populate: keyType in request null; default to RSA");
+            }
+
+            boolean isEC = false;
+            String keySizeCurveStr = request.getExtDataInString("keySize");
+
+            if (keyType.contentEquals("RSA")) {
+                if (keySizeCurveStr != null && !keySizeCurveStr.isEmpty()) {
+                    logger.debug("ServerKeygenUserKeyDefault: populate: keySize in request: " + keySizeCurveStr);
+                    keySize = Integer.parseInt(keySizeCurveStr);
+                } else {
+                    logger.debug("ServerKeygenUserKeyDefault: populate: keySize in request null;  default to" + keySize);
+                }
+                // Do things when RSA is selected
+            } else if (keyType.contentEquals("EC")) {
+                isEC = true;
+                // TODO: dmoluguw: Fix the following to generate right Key ECC keys
+
+                if (keySizeCurveStr != null && !keySizeCurveStr.isEmpty()) {
+                    logger.debug("ServerKeygenUserKeyDefault: populate: keyCurve in request: " + keySizeCurveStr);
+                    curveName = keySizeCurveStr;
+                } else {
+                    logger.debug("ServerKeygenUserKeyDefault: populate: keySize in request null;  default to" + curveName);
+                }
+                // Do things when EC is selected
+            } else {
+                throw new Exception("Unsupported keyType: " + keyType);
+            }
+            request.setExtData(IRequest.KEY_GEN_ALGORITHM, keyType);
+            if(keyType.contentEquals("RSA")) {
+                request.setExtData(IRequest.KEY_GEN_SIZE, keySize);
+            }
+            else if (keyType.contentEquals("EC")) {
+                // TODO: Check whether IRequest.KEY_GEN_SIZE can accept string value
+                request.setExtData(IRequest.KEY_GEN_SIZE, curveName);
+            }
+
+            /*
+             * DO NOT REMOVE
+             * currently, it is necessary to  put in a static placeholder fake
+             * key here to prevent issue; The fake key will be replaced later
+             * once KRA generates the real keys
+             *
+             * TODO: perhaps find out how to get around not breaking
+             * the code without fake keys
+             */
+
+            String pubKeyStr = "";
```

Comment:
In the default case of the following two switch statements, should we be throwing an exception instead of letting it go?

Reply:
will do.